### DISCUSSION TITLE
Discord oauth fails to complete

### DISCUSSION THREAD

**Discussion by theolas** [requarks/wiki#5415] [Error / Bug Report] [answered]
I've been trying to setup discord authentication. Followed all the [instructions](https://docs.requarks.io/auth/discord). It seems to fail after the "an external application wants to access your discord account" 

![image](https://user-images.githubusercontent.com/8877351/174438810-4769838b-5741-4c83-8d98-7b36047b45bf.png)

and shows the wiki.js "something went wrong" page for the callback url. 

![image](https://user-images.githubusercontent.com/8877351/174438852-d7ce0317-e10b-45b8-8bd0-42de1d072c1a.png)

tried to troubleshoot on discord, redid the wiki and discord sides of the config, still no dice. 

https://discord.com/channels/911122923396431892/912820720193634364/987593833510490182

plis help

Installed on a Digital Ocean Droplet

**Comment by ldpr** [comment]
@theolas are you also using MariaDB?

I get an error at the same point, but I have an error message

![](https://crowdfile.net/o0QQ41Lz.png)

**Reply by theolas** [reply]
I'm using whatever the droplet installer came with.

**Comment by julianalsemmani** [comment]
Can we see how you've setup the config for the Discord Auth? Just hide the API keys.

**Comment by theolas** [comment]
![image](https://user-images.githubusercontent.com/8877351/177434589-0c4e104c-3fb4-4faa-8de4-b77ef096699e.png)
![image](https://user-images.githubusercontent.com/8877351/177434609-c2dd7f78-ee7e-4465-b252-24f7f1ef745c.png)

You can check out our convo in the discord server: 
https://discord.com/channels/911122923396431892/912820720193634364/987593833510490182

**Comment by zigazajc007** [comment]
Did anyone found the solution to this problem?

It still persist in the latest Wiki JS version.

**Comment by minestarnl** [comment]
I also have this issue, any updates?

**Comment by cgvictor** [comment]
Same situation persists (nov.2023), **Discord auth is not working in 2.5.300**

To be precise, the callback redirect to jswiki from DS fails with 500
this one: https://jswiki-domain/login/******/callback?code=*****

**Comment by cgvictor** [comment, marked as answer]
Guys, after a bit of debug got this **SOLVED**.
Actually, passport-discord fails with TokenError (shown in dev mode), and the reason is: **you probably are using publickey instead of application secret**. As I did.

So, steps to "fix" this on 2.5.300 stable:
- re-create the application secret in Discord application area, your app -> OAuth2 -> Reset secret
- inside wiki Admin area, put the secret into app secret field, then Save changes

Crosscheck: **your secret should be alphanumeric, short about 30 chars, NOT the long, hash-like 65 chars**.

I admit that Discord UI is a bit confusing on this one, to give you (appid+publickey) pair instead of (app+secret) as widely used. 
But OK, things are back on track. Hope this memo helps someone. 

@zigazajc007 @minestarnl @theolas take a look ^
@NGPixel thanks for the dev environment! I'm not a nodejs developer, but have deployed a dev env locally in no time thanks to your contrib instruction, so debug option was a huge help for me

`TokenError
    at OAuth2Strategy.parseErrorResponse (/wiki/node_modules/passport-discord/node_modules/passport-oauth2/lib/strategy.js:358:12)
    at OAuth2Strategy._createOAuthError (/wiki/node_modules/passport-discord/node_modules/passport-oauth2/lib/strategy.js:405:16)
    at /wiki/node_modules/passport-discord/node_modules/passport-oauth2/lib/strategy.js:175:45`

**Reply by Netherwhal** [reply]
Thanks for confirming. In my case I had an accidental whitespace at the beginning when copying the secret. Worthwhile double checking.
